src/bounce_solution/bounce_solution.cpp

```diff
@@ -0,0 +1,1256 @@
+    std::vector<double> TrueVacuum = TransformIntoOptimalDiscreteSymmetry(
+        phase_pair.true_phase.Get(T).point);
+    std::vector<double> FalseVacuum = phase_pair.false_phase.Get(T).point;
+    std::function<double(std::vector<double>)> V = [&](std::vector<double> vev)
+    {
+      // Potential wrapper
+      return modelPointer->VEff(modelPointer->MinimizeOrderVEV(vev), T);
+    };
+    std::vector<std::vector<double>> path = {TrueVacuum, FalseVacuum};
+
+    BounceActionInt bc(
+        path, TrueVacuum, FalseVacuum, V, T, MaxPathIntegrations);
+    bc.CalculateAction();
+    if (bc.action / T > 0)
+    {
+      SolutionList.push_back(bc);
+    }
+  }
+}
+
+void BounceSolution::GWSecondaryScan()
+{
+  if (SolutionList.size() == 0)
+  {
+    Logger::Write(LoggingLevel::BounceDetailed,
+                  "No solution was found during the initial scan.\n Abort!\n");
+    // Failed to find the solution
+    status_bounce_sol = "failure";
+    return;
+  }
+  else if (SolutionList.size() == 1)
+  {
+    Logger::Write(
+        LoggingLevel::BounceDetailed,
+        "Only one solution was found. Searching near what was found\n");
+    CalculateActionAt(SolutionList[0].T - 0.01);
+    if (SolutionList.size() == 1)
+    {
+      // Failed to find the solution
+      SetBounceSol();
+      return;
+    };
+  }
+
+  GWScanTowardsLowAction();
+  GWScanTowardsHighAction();
+
+  std::size_t NumOfSol = SolutionList.size();
+
+  for (int i = 0; i < 2; i++)
+  {
+    std::vector<double> nextTList;
+
+    for (std::size_t sol = 0; sol < SolutionList.size() - 1; sol++)
+    {
+      if (SolutionList[sol].action / SolutionList[sol].T > 200) continue;
+      if (SolutionList[sol + 1].action / SolutionList[sol + 1].T < 50) continue;
+
+      double NumOfT =
+          ceil((SolutionList[sol + 1].action / SolutionList[sol + 1].T -
+                SolutionList[sol].action / SolutionList[sol].T) /
+               20);
+
+      double dT = (SolutionList[sol + 1].T - SolutionList[sol].T) / NumOfT;
+      for (int it_T = 1; it_T < NumOfT; it_T++)
+      {
+        nextTList.push_back(SolutionList[sol].T + it_T * dT);
+      }
+    }
+
+    for (auto j : nextTList)
+    {
+      CalculateActionAt(j);
+    }
+
+    if (NumOfSol == SolutionList.size()) break;
+    NumOfSol = SolutionList.size();
+  }
+  GWScanTowardsLowAction();
+  GWScanTowardsHighAction();
+  SetBounceSol();
+}
+
+void BounceSolution::GWScanTowardsHighAction()
+{
+  for (int i = 0;
+       i <=
+       1. + (200 - SolutionList.back().action / SolutionList.back().T) / 5.;
+       i++)
+  {
+    if (SolutionList.back().action / SolutionList.back().T > 200) break;
+
+    // Try to calculate until s3/T = 200
+    double t1            = SolutionList[SolutionList.size() - 2].T;
+    double t2            = SolutionList[SolutionList.size() - 1].T;
+    double s1            = SolutionList[SolutionList.size() - 2].action / t1;
+    double s2            = SolutionList[SolutionList.size() - 1].action / t2;
```

Comment:
Add check for size < 2

Reply:
Already there. If ```SolutionList.size() == 0```or ```SolutionList.size() == 1``` we abort the calculation before.